blocks/api/factory.js

```diff
@@ -17,3 +23,32 @@ export function createBlock( blockType, attributes = {} ) {
 		attributes
 	};
 }
+
+/**
+ * Switch Block Type and returns the updated block
+ *
+ * @param  {Object} block      Block object
+ * @param  {string} blockType  BlockType
+ * @return {Object?}           Block object
+ */
+export function switchToBlockType( block, blockType ) {
+	const destinationSettings = getBlockSettings( blockType );
+	const sourceSettings = getBlockSettings( block.blockType );
+	const transformationsFrom = get( destinationSettings, 'transforms.from', [] );
+	const transformationsTo = get( sourceSettings, 'transforms.to', [] );
+
+	// Find the from transformation
+	const transformation =
+		transformationsFrom.find( t => t.blocks.indexOf( block.blockType ) !== -1 ) ||
+		transformationsTo.find( t => t.blocks.indexOf( blockType ) !== -1 );
```

Comment:
Yes, I had to make a choice. I remember vaguely discussing this while implementing the per-block prototype and choosing the `from` transformation as the default but don't remember the arguments though.

Thinking about it more, seems more logic to give priority to the `to` transformation.